api/src/database/migrations/20211101140003-add-set-null-to-users-created-fields.ts

```diff
@@ -0,0 +1,21 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+	await knex.schema.alterTable('directus_activity', (table) => {
+		table.foreign('user').references('directus_users.id').onDelete('SET NULL');
+	});
+	await knex.schema.alterTable('directus_files', (table) => {
+		table.dropForeign('uploaded_by');
+		table.dropForeign('modified_by');
+	});
+	return await knex.schema.alterTable('directus_files', (table) => {
+		table.foreign('uploaded_by').references('directus_users.id').onDelete('SET NULL');
+		table.foreign('modified_by').references('directus_users.id').onDelete('SET NULL');
+	});
+}
+
+export async function down(knex: Knex): Promise<void> {
+	await knex.schema.alterTable('directus_files', (table) => {
+		table.dropColumn('uploaded_by');
```

Comment:
Is this supposed to drop the column?

Reply:
No. Just the constraint so that I could add the new constraint with the appropriate .onDelete(). I was initially going to just say:
`table.foreign('uploaded_by').references('directus_users.id').onDelete('SET NULL').alter();`
But I found that to change the constraint like that I had to first drop the constraint and add a new one. I think dropColumn() will be a lot more destructive than just dropping the constraint.

Reply:
Oh I see what you mean.